### PR TITLE
[FSSDK-9776] handle duplicate experiment key

### PR DESCRIPTION
**Pull request by muzahidul-opti**
## Summary
When duplicate experiment key found in the data file, SDK returns correct experiment map in  OptimizelyConfig

- Keep latest experiment key, found in the datafile experiment list
- Add warning log "Duplicate keys found in datafile: {key-name}"

## Test plan
- Add test case for OptimizelyConfig with duplicate key

## Issues
- [FSSDK-9776](https://jira.sso.episerver.net/browse/FSSDK-9776)

### CODE REVIEW

**Review by mikechu-optimizely — DISMISSED** [review]
Some thoughts but LGTM.

File: Tests/OptimizelyTests-APIs/OptimizelyClientTests_OptimizelyConfig.swift [hunk 1]
```diff
@@ -257,6 +257,67 @@ class OptimizelyClientTests_OptimizelyConfig: XCTestCase {
...
+		optimizely = OptimizelyClient(
+			sdkKey: "demo_key"
+		)
+		
+		optimizely.config = projectConfig
+				
+		let optimizelyExpMap: [String: OptimizelyExperiment] = try! optimizely.getOptimizelyConfig().experimentsMap
+		
+		XCTAssertEqual(optimizelyExpMap.count, 1)
+		XCTAssertEqual(optimizelyExpMap["duplicate_key"]?.id, "10005")
```
[the remark this answers is not in the document]
Comment (muzahidul-opti) [reply Tests/OptimizelyTests-APIs/OptimizelyClientTests_OptimizelyConfig.swift:320]: Logger test added

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good. Can you check why your code format/styles are different from existing codes?

**Comment by mikechu-optimizely** [comment]
Can I use this as a reference for doing other language SDK updates?

**Comment by muzahidul-opti** [comment]
> Can I use this as a reference for doing other language SDK updates?

Sure

**Comment by mikechu-optimizely** [comment]
Thanks for the lint fixes. This helps my need for meticulous code.  😄 

Also thanks for the reference implementation for me to copy 😁
